pilot/cmd/pilot-agent/main.go

```diff
@@ -334,7 +341,7 @@ var (
 					localHostAddr = localHostIPv6
 				}
 				tokenManager := tokenmanager.CreateTokenManager(tokenManagerPlugin,
-					tokenmanager.Config{TrustDomain: trustDomain})
+					tokenmanager.Config{CredFetcher: credFetcher, TrustDomain: secOpts.TrustDomain})
```

Comment:
Don't we have trustdomain in ProxyConfig ? How does it get set in secOpts  ?

Reply:
It's set by environment variable. It's the existing code.